src/GapicClientTrait.php

```diff
@@ -183,6 +184,14 @@ private function buildClientOptions(array $options)
             $options['apiEndpoint'] = $this->pluck('serviceAddress', $options, false);
         }
 
+        // Use default scopes if an API endpoint is set to ensure the "audience"
+        // does not conflict with the custom endpoint.
+        if (empty($options['credentialsConfig']['scopes'])
+                && $options['apiEndpoint'] != $defaultOptions['apiEndpoint']
+                && isset($options['defaultScopes'])) {
+            $options['credentialsConfig']['scopes'] = $options['defaultScopes'];
```

Comment:
`defaultScopes` should be passed for GCE credentials and UserRefresh credentials, so we most likely need to move this logic into `google/auth`